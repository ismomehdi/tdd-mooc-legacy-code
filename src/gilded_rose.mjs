export class Item {
  constructor(name, sellIn, quality) {
    this.name = name;
    this.sellIn = sellIn;
    this.quality = quality;
  }
}

export class Shop {
  constructor(items = []) {
    this.items = items;
  }

  updateQuality() {
    this.items.forEach((item) => {
      if (item.name === "Aged Brie") {
        this.#updateAgedBrie(item);
      } else if (item.name === "Backstage passes to a TAFKAL80ETC concert") {
        this.#updateBackstagePass(item);
      } else if (item.name === "Conjured") {
        this.#updateConjuredItem(item);
      } else if (item.name !== "Sulfuras, Hand of Ragnaros") {
        this.#updateRegularItem(item);
      }
    });
    return this.items;
  }

  #updateConjuredItem(item) {
    if (item.quality > 0) {
      item.quality -= 2;
    }
    item.sellIn = item.sellIn - 1;
    if (item.sellIn < 0) {
      if (item.quality > 0) {
        item.quality -= 2;
      }
    }
  }

  #updateRegularItem(item) {
    if (item.quality > 0) {
      item.quality -= 1;
    }
    item.sellIn = item.sellIn - 1;
    if (item.sellIn < 0) {
      if (item.quality > 0) {
        item.quality -= 1;
      }
    }
  }

  #updateBackstagePass(item) {
    if (item.quality < 50) {
      item.quality += 1;
      if (item.sellIn < 11 && item.quality < 50) {
        item.quality += 1;
        if (item.sellIn < 6) {
          item.quality += 1;
        }
      }
    }
    item.sellIn -= 1;
    if (item.sellIn < 0) {
      item.quality = item.quality - item.quality;
    }
  }

  #updateAgedBrie(item) {
    if (item.quality < 50) {
      item.quality += 1;
    }

    item.sellIn -= 1;
    if (item.sellIn < 0 && item.quality < 50) {
      item.quality += 1;
    }
  }
}
