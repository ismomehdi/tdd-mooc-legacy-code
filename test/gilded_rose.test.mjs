import { describe, test } from "vitest";
import { expect } from "chai";
import { Item, Shop } from "../src/gilded_rose.mjs";

describe("Gilded Rose", () => {
  test("foo", () => {
    const gildedRose = new Shop([
      new Item("foo", 0, 0),
      new Item("Backstage passes to a TAFKAL80ETC concert", 0, 0),
      new Item("Backstage passes to a TAFKAL80ETC concert", 12, 0),
    ]);
    const items = gildedRose.updateQuality();

    const expected = [
      { name: "foo", sellIn: -1 },
      { name: "Backstage passes to a TAFKAL80ETC concert", sellIn: -1 },
      { name: "Backstage passes to a TAFKAL80ETC concert", sellIn: 11 },
    ];

    for (let i = 0; i < items.length; i++) {
      expect(items[i].name).to.equal(expected[i].name);
      expect(items[i].sellIn).to.equal(expected[i].sellIn);
    }
  });
});
