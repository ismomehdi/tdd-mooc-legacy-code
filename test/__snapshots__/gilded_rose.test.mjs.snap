// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`__default_name_ 1`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 2`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 3`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 0,
}
`;

exports[`__default_name_ 4`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 1,
}
`;

exports[`__default_name_ 5`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 5,
}
`;

exports[`__default_name_ 6`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 9,
}
`;

exports[`__default_name_ 7`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 10,
}
`;

exports[`__default_name_ 8`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 9`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 10`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 0,
}
`;

exports[`__default_name_ 11`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 1,
}
`;

exports[`__default_name_ 12`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 5,
}
`;

exports[`__default_name_ 13`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 9,
}
`;

exports[`__default_name_ 14`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 10,
}
`;

exports[`__default_name_ 15`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 16`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 17`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 0,
}
`;

exports[`__default_name_ 18`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 1,
}
`;

exports[`__default_name_ 19`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 5,
}
`;

exports[`__default_name_ 20`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 9,
}
`;

exports[`__default_name_ 21`] = `
Item {
  "name": "foo",
  "quality": 0,
  "sellIn": 10,
}
`;

exports[`__default_name_ 22`] = `
Item {
  "name": "foo",
  "quality": 47,
  "sellIn": -2,
}
`;

exports[`__default_name_ 23`] = `
Item {
  "name": "foo",
  "quality": 47,
  "sellIn": -1,
}
`;

exports[`__default_name_ 24`] = `
Item {
  "name": "foo",
  "quality": 48,
  "sellIn": 0,
}
`;

exports[`__default_name_ 25`] = `
Item {
  "name": "foo",
  "quality": 48,
  "sellIn": 1,
}
`;

exports[`__default_name_ 26`] = `
Item {
  "name": "foo",
  "quality": 48,
  "sellIn": 5,
}
`;

exports[`__default_name_ 27`] = `
Item {
  "name": "foo",
  "quality": 48,
  "sellIn": 9,
}
`;

exports[`__default_name_ 28`] = `
Item {
  "name": "foo",
  "quality": 48,
  "sellIn": 10,
}
`;

exports[`__default_name_ 29`] = `
Item {
  "name": "foo",
  "quality": 48,
  "sellIn": -2,
}
`;

exports[`__default_name_ 30`] = `
Item {
  "name": "foo",
  "quality": 48,
  "sellIn": -1,
}
`;

exports[`__default_name_ 31`] = `
Item {
  "name": "foo",
  "quality": 49,
  "sellIn": 0,
}
`;

exports[`__default_name_ 32`] = `
Item {
  "name": "foo",
  "quality": 49,
  "sellIn": 1,
}
`;

exports[`__default_name_ 33`] = `
Item {
  "name": "foo",
  "quality": 49,
  "sellIn": 5,
}
`;

exports[`__default_name_ 34`] = `
Item {
  "name": "foo",
  "quality": 49,
  "sellIn": 9,
}
`;

exports[`__default_name_ 35`] = `
Item {
  "name": "foo",
  "quality": 49,
  "sellIn": 10,
}
`;

exports[`__default_name_ 36`] = `
Item {
  "name": "foo",
  "quality": 49,
  "sellIn": -2,
}
`;

exports[`__default_name_ 37`] = `
Item {
  "name": "foo",
  "quality": 49,
  "sellIn": -1,
}
`;

exports[`__default_name_ 38`] = `
Item {
  "name": "foo",
  "quality": 50,
  "sellIn": 0,
}
`;

exports[`__default_name_ 39`] = `
Item {
  "name": "foo",
  "quality": 50,
  "sellIn": 1,
}
`;

exports[`__default_name_ 40`] = `
Item {
  "name": "foo",
  "quality": 50,
  "sellIn": 5,
}
`;

exports[`__default_name_ 41`] = `
Item {
  "name": "foo",
  "quality": 50,
  "sellIn": 9,
}
`;

exports[`__default_name_ 42`] = `
Item {
  "name": "foo",
  "quality": 50,
  "sellIn": 10,
}
`;

exports[`__default_name_ 43`] = `
Item {
  "name": "Aged Brie",
  "quality": 1,
  "sellIn": -2,
}
`;

exports[`__default_name_ 44`] = `
Item {
  "name": "Aged Brie",
  "quality": 1,
  "sellIn": -1,
}
`;

exports[`__default_name_ 45`] = `
Item {
  "name": "Aged Brie",
  "quality": 0,
  "sellIn": 0,
}
`;

exports[`__default_name_ 46`] = `
Item {
  "name": "Aged Brie",
  "quality": 0,
  "sellIn": 1,
}
`;

exports[`__default_name_ 47`] = `
Item {
  "name": "Aged Brie",
  "quality": 0,
  "sellIn": 5,
}
`;

exports[`__default_name_ 48`] = `
Item {
  "name": "Aged Brie",
  "quality": 0,
  "sellIn": 9,
}
`;

exports[`__default_name_ 49`] = `
Item {
  "name": "Aged Brie",
  "quality": 0,
  "sellIn": 10,
}
`;

exports[`__default_name_ 50`] = `
Item {
  "name": "Aged Brie",
  "quality": 2,
  "sellIn": -2,
}
`;

exports[`__default_name_ 51`] = `
Item {
  "name": "Aged Brie",
  "quality": 2,
  "sellIn": -1,
}
`;

exports[`__default_name_ 52`] = `
Item {
  "name": "Aged Brie",
  "quality": 1,
  "sellIn": 0,
}
`;

exports[`__default_name_ 53`] = `
Item {
  "name": "Aged Brie",
  "quality": 1,
  "sellIn": 1,
}
`;

exports[`__default_name_ 54`] = `
Item {
  "name": "Aged Brie",
  "quality": 1,
  "sellIn": 5,
}
`;

exports[`__default_name_ 55`] = `
Item {
  "name": "Aged Brie",
  "quality": 1,
  "sellIn": 9,
}
`;

exports[`__default_name_ 56`] = `
Item {
  "name": "Aged Brie",
  "quality": 1,
  "sellIn": 10,
}
`;

exports[`__default_name_ 57`] = `
Item {
  "name": "Aged Brie",
  "quality": 3,
  "sellIn": -2,
}
`;

exports[`__default_name_ 58`] = `
Item {
  "name": "Aged Brie",
  "quality": 3,
  "sellIn": -1,
}
`;

exports[`__default_name_ 59`] = `
Item {
  "name": "Aged Brie",
  "quality": 2,
  "sellIn": 0,
}
`;

exports[`__default_name_ 60`] = `
Item {
  "name": "Aged Brie",
  "quality": 2,
  "sellIn": 1,
}
`;

exports[`__default_name_ 61`] = `
Item {
  "name": "Aged Brie",
  "quality": 2,
  "sellIn": 5,
}
`;

exports[`__default_name_ 62`] = `
Item {
  "name": "Aged Brie",
  "quality": 2,
  "sellIn": 9,
}
`;

exports[`__default_name_ 63`] = `
Item {
  "name": "Aged Brie",
  "quality": 2,
  "sellIn": 10,
}
`;

exports[`__default_name_ 64`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": -2,
}
`;

exports[`__default_name_ 65`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": -1,
}
`;

exports[`__default_name_ 66`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 0,
}
`;

exports[`__default_name_ 67`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 1,
}
`;

exports[`__default_name_ 68`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 5,
}
`;

exports[`__default_name_ 69`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 9,
}
`;

exports[`__default_name_ 70`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 10,
}
`;

exports[`__default_name_ 71`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": -2,
}
`;

exports[`__default_name_ 72`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": -1,
}
`;

exports[`__default_name_ 73`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 0,
}
`;

exports[`__default_name_ 74`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 1,
}
`;

exports[`__default_name_ 75`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 5,
}
`;

exports[`__default_name_ 76`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 9,
}
`;

exports[`__default_name_ 77`] = `
Item {
  "name": "Aged Brie",
  "quality": 50,
  "sellIn": 10,
}
`;

exports[`__default_name_ 78`] = `
Item {
  "name": "Aged Brie",
  "quality": 51,
  "sellIn": -2,
}
`;

exports[`__default_name_ 79`] = `
Item {
  "name": "Aged Brie",
  "quality": 51,
  "sellIn": -1,
}
`;

exports[`__default_name_ 80`] = `
Item {
  "name": "Aged Brie",
  "quality": 51,
  "sellIn": 0,
}
`;

exports[`__default_name_ 81`] = `
Item {
  "name": "Aged Brie",
  "quality": 51,
  "sellIn": 1,
}
`;

exports[`__default_name_ 82`] = `
Item {
  "name": "Aged Brie",
  "quality": 51,
  "sellIn": 5,
}
`;

exports[`__default_name_ 83`] = `
Item {
  "name": "Aged Brie",
  "quality": 51,
  "sellIn": 9,
}
`;

exports[`__default_name_ 84`] = `
Item {
  "name": "Aged Brie",
  "quality": 51,
  "sellIn": 10,
}
`;

exports[`__default_name_ 85`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 86`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 87`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 2,
  "sellIn": 0,
}
`;

exports[`__default_name_ 88`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 2,
  "sellIn": 1,
}
`;

exports[`__default_name_ 89`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 1,
  "sellIn": 5,
}
`;

exports[`__default_name_ 90`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 1,
  "sellIn": 9,
}
`;

exports[`__default_name_ 91`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": 10,
}
`;

exports[`__default_name_ 92`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 93`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 94`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 3,
  "sellIn": 0,
}
`;

exports[`__default_name_ 95`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 3,
  "sellIn": 1,
}
`;

exports[`__default_name_ 96`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 2,
  "sellIn": 5,
}
`;

exports[`__default_name_ 97`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 2,
  "sellIn": 9,
}
`;

exports[`__default_name_ 98`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 1,
  "sellIn": 10,
}
`;

exports[`__default_name_ 99`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 100`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 101`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 4,
  "sellIn": 0,
}
`;

exports[`__default_name_ 102`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 4,
  "sellIn": 1,
}
`;

exports[`__default_name_ 103`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 3,
  "sellIn": 5,
}
`;

exports[`__default_name_ 104`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 3,
  "sellIn": 9,
}
`;

exports[`__default_name_ 105`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 2,
  "sellIn": 10,
}
`;

exports[`__default_name_ 106`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 107`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 108`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 0,
}
`;

exports[`__default_name_ 109`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 1,
}
`;

exports[`__default_name_ 110`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 5,
}
`;

exports[`__default_name_ 111`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 9,
}
`;

exports[`__default_name_ 112`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 10,
}
`;

exports[`__default_name_ 113`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 114`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 115`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 0,
}
`;

exports[`__default_name_ 116`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 1,
}
`;

exports[`__default_name_ 117`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 5,
}
`;

exports[`__default_name_ 118`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 9,
}
`;

exports[`__default_name_ 119`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 50,
  "sellIn": 10,
}
`;

exports[`__default_name_ 120`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -2,
}
`;

exports[`__default_name_ 121`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 122`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 51,
  "sellIn": 0,
}
`;

exports[`__default_name_ 123`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 51,
  "sellIn": 1,
}
`;

exports[`__default_name_ 124`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 51,
  "sellIn": 5,
}
`;

exports[`__default_name_ 125`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 51,
  "sellIn": 9,
}
`;

exports[`__default_name_ 126`] = `
Item {
  "name": "Backstage passes to a TAFKAL80ETC concert",
  "quality": 51,
  "sellIn": 10,
}
`;

exports[`__default_name_ 127`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 128`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 0,
}
`;

exports[`__default_name_ 129`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 1,
}
`;

exports[`__default_name_ 130`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 2,
}
`;

exports[`__default_name_ 131`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 6,
}
`;

exports[`__default_name_ 132`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 10,
}
`;

exports[`__default_name_ 133`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 11,
}
`;

exports[`__default_name_ 134`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": -1,
}
`;

exports[`__default_name_ 135`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 0,
}
`;

exports[`__default_name_ 136`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 1,
}
`;

exports[`__default_name_ 137`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 2,
}
`;

exports[`__default_name_ 138`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 6,
}
`;

exports[`__default_name_ 139`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 10,
}
`;

exports[`__default_name_ 140`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 0,
  "sellIn": 11,
}
`;

exports[`__default_name_ 141`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 1,
  "sellIn": -1,
}
`;

exports[`__default_name_ 142`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 1,
  "sellIn": 0,
}
`;

exports[`__default_name_ 143`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 1,
  "sellIn": 1,
}
`;

exports[`__default_name_ 144`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 1,
  "sellIn": 2,
}
`;

exports[`__default_name_ 145`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 1,
  "sellIn": 6,
}
`;

exports[`__default_name_ 146`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 1,
  "sellIn": 10,
}
`;

exports[`__default_name_ 147`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 1,
  "sellIn": 11,
}
`;

exports[`__default_name_ 148`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 49,
  "sellIn": -1,
}
`;

exports[`__default_name_ 149`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 49,
  "sellIn": 0,
}
`;

exports[`__default_name_ 150`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 49,
  "sellIn": 1,
}
`;

exports[`__default_name_ 151`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 49,
  "sellIn": 2,
}
`;

exports[`__default_name_ 152`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 49,
  "sellIn": 6,
}
`;

exports[`__default_name_ 153`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 49,
  "sellIn": 10,
}
`;

exports[`__default_name_ 154`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 49,
  "sellIn": 11,
}
`;

exports[`__default_name_ 155`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 50,
  "sellIn": -1,
}
`;

exports[`__default_name_ 156`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 50,
  "sellIn": 0,
}
`;

exports[`__default_name_ 157`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 50,
  "sellIn": 1,
}
`;

exports[`__default_name_ 158`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 50,
  "sellIn": 2,
}
`;

exports[`__default_name_ 159`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 50,
  "sellIn": 6,
}
`;

exports[`__default_name_ 160`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 50,
  "sellIn": 10,
}
`;

exports[`__default_name_ 161`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 50,
  "sellIn": 11,
}
`;

exports[`__default_name_ 162`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 51,
  "sellIn": -1,
}
`;

exports[`__default_name_ 163`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 51,
  "sellIn": 0,
}
`;

exports[`__default_name_ 164`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 51,
  "sellIn": 1,
}
`;

exports[`__default_name_ 165`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 51,
  "sellIn": 2,
}
`;

exports[`__default_name_ 166`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 51,
  "sellIn": 6,
}
`;

exports[`__default_name_ 167`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 51,
  "sellIn": 10,
}
`;

exports[`__default_name_ 168`] = `
Item {
  "name": "Sulfuras, Hand of Ragnaros",
  "quality": 51,
  "sellIn": 11,
}
`;
